extensions/kubernetes-client/spi/pom.xml

```diff
@@ -44,6 +44,10 @@
             <groupId>io.fabric8</groupId>
             <artifactId>kubernetes-httpclient-vertx</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.dajudge.kindcontainer</groupId>
+            <artifactId>kindcontainer</artifactId>
+        </dependency>
```

Comment:
We definitely don't want to add this dependency to the SPI module

Reply:
You could possibly create an ArgocdContainer object, using host and port.

Reply:
> You could possibly create an ArgocdContainer object, using host and port.

Developing an Argocd image able to launch a kubernetes cluster where argocd is deployed will require a lot of effort like also to maintain it. 

This is why the approach where we rely on the existing kube cluster created by the kubernetes-client is a better solution as we must only handle the part to install argocd: https://argo-cd.readthedocs.io/en/stable/getting_started/#1-install-argo-cd

Are you thinking about something else ?